From: Mario Sanchez Prada <mario@endlessm.com>
Date: Thu, 7 Dec 2017 16:16:29 +0000
Subject: request: Stop hardcoding response codes, use an enum instead

Define a new enum XdgDesktopPortalResponseEnum to use in the few
places where we are currently hardcoding them (normally, we simply
propagate the response code coming from the backend), and use it.

Bug: https://github.com/flatpak/xdg-desktop-portal/issues/134
Origin: upstream, 0.10, commit:2799a4558e4154cfa218247c6b129c3cfc580cb0
---
 src/device.c   | 2 +-
 src/open-uri.c | 8 ++++++--
 src/request.h  | 6 ++++++
 3 files changed, 13 insertions(+), 3 deletions(-)

diff --git a/src/device.c b/src/device.c
index c99f576..2234bfd 100644
--- a/src/device.c
+++ b/src/device.c
@@ -267,7 +267,7 @@ handle_access_microphone_in_thread (GTask *task,
 
       g_variant_builder_init (&results, G_VARIANT_TYPE_VARDICT);
       xdp_request_emit_response (XDP_REQUEST (request),
-                                 allowed ? 0 : 1,
+                                 allowed ? XDG_DESKTOP_PORTAL_RESPONSE_SUCCESS : XDG_DESKTOP_PORTAL_RESPONSE_CANCELLED,
                                  g_variant_builder_end (&results));
       request_unexport (request);
     }
diff --git a/src/open-uri.c b/src/open-uri.c
index a13ce98..c80f3cd 100644
--- a/src/open-uri.c
+++ b/src/open-uri.c
@@ -485,7 +485,9 @@ handle_open_in_thread_func (GTask *task,
           if (request->exported)
             {
               g_variant_builder_init (&opts_builder, G_VARIANT_TYPE_VARDICT);
-              xdp_request_emit_response (XDP_REQUEST (request), 2, g_variant_builder_end (&opts_builder));
+              xdp_request_emit_response (XDP_REQUEST (request),
+                                         XDG_DESKTOP_PORTAL_RESPONSE_OTHER,
+                                         g_variant_builder_end (&opts_builder));
               request_unexport (request);
             }
           return;
@@ -502,7 +504,9 @@ handle_open_in_thread_func (GTask *task,
           if (request->exported)
             {
               g_variant_builder_init (&opts_builder, G_VARIANT_TYPE_VARDICT);
-              xdp_request_emit_response (XDP_REQUEST (request), 2, g_variant_builder_end (&opts_builder));
+              xdp_request_emit_response (XDP_REQUEST (request),
+                                         XDG_DESKTOP_PORTAL_RESPONSE_OTHER,
+                                         g_variant_builder_end (&opts_builder));
               request_unexport (request);
             }
           return;
diff --git a/src/request.h b/src/request.h
index 9e829ef..7319330 100644
--- a/src/request.h
+++ b/src/request.h
@@ -24,6 +24,12 @@
 #include "xdp-dbus.h"
 #include "xdp-impl-dbus.h"
 
+typedef enum {
+  XDG_DESKTOP_PORTAL_RESPONSE_SUCCESS = 0,
+  XDG_DESKTOP_PORTAL_RESPONSE_CANCELLED,
+  XDG_DESKTOP_PORTAL_RESPONSE_OTHER
+} XdgDesktopPortalResponseEnum;
+
 typedef struct _Request Request;
 typedef struct _RequestClass RequestClass;
 
